Completed Part 1 of the Tutorial Dapp (#82)
MetaMask/metamask-docs: docs/guide/create-dapp.md

# Create A Simple Dapp
::: tip 
We will be building this [app](https://metamask.github.io/test-dapp/)
:::

## Project Setup

> Before you setup make sure you've visited and gone through our [Getting Started Guide](http://localhost:8080/guide/getting-started.html#getting-started)

Reviewer: ```suggestion
Before you setup make sure you've visited and gone through our [Getting Started Guide](/guide/getting-started.html#getting-started)
```

Reviewer: ^is this for the dapp getting started guide, or the general getting started with metamask guide?

Author: This is the general getting started guide.

Author: I wanted to go back and make the general getting started more "hands on" than what is currently in the docs

Make sure you have:

> 1. Have the [MetaMask Extendsion](https://metamask.io/download.html) downlaoded.

Reviewer: ```suggestion
1. Have the [MetaMask Extension](https://metamask.io/download.html) downloaded.
```

3. Clone/Download the [project files](https://github.com/BboyAkers/simple-dapp-tutorial) from Github.
4. Have your favorite Text Editor or IDE installed. I personally like [Visual Studio Code](https://code.visualstudio.com/)

### Open Project Folder
Open the poejct folder. Navigate to start->index.html, and look at the comment stating part 1. We will be using/buidling off of this entire section for the first part of the tutorial.

### Install Dependencies
Open a terminal and make sure your terminal is inside the base diretory of the start/ folder. Inside the folder the files should look like this:

```
.
├─ index.html
├─ contract.js
├─ metamask.css
├─ package.json
└─ README.md
```
You'll have some more files but that's nothing worry about!

Open your terminal and navigate into the start folder. In this folder run:
``` bash
npm install
```
This will install all the neccessary dependencies we'll need for our project. This will have created a node_modules/ folder where all the dependencies are stored.

Next run:
``` bash
npm run serve
```

Navigate to `http://localhost:9011`

## Basic Action(Part 1)

Now let's navigate into the contract.js file inside you start folder.

> Your file should look something like this. Don't worry about line 1-31.

Reviewer: ```suggestion
Your file should look something like this. Don't worry about lines 1-31.
```

``` javascript
const forwarderOrigin = 'http://localhost:9010'

const initialize = () => {
  //You will start here 
}
window.addEventListener('DOMContentLoaded', initialize)
```

> As you can see here, as soon as the content in the DOM is loaded we are calling our initialized function. Now before we start writing any code we need to see what's on our task list for the first part of this app.

Reviewer: ```suggestion
As you can see here, as soon as the content in the DOM is loaded we are calling our initialize function. Now before we start writing any code we need to see what's on our task list for the first part of this app.
```

What we'll cover in part one:
  
- [Connecting to the MetaMask Wallet](/guide/create-dapp.html#connecting-to-the-metamask-wallet)
- See our eth_accounts result
- Display our network number
- Display our ChainId
- Display our Accounts

### Connecting to the MetaMask Wallet

The first thing we need to do in our Dapp is to connect to our MetaMask Wallet.
1. We need to create a function to see if the MetaMask Chrome extension is installed
2. If MetaMask is not installed we:
   1. Change our `connectButton` to `Click here to install MetaMask`
   2. When clicking that button it should take us to a page that will allow us to install the extension
   3. Disable the button
3. If MetaMask is installed we:
   1. Change our `connectButton` to `Connect`
   2. When clicking that button it should allow us to connect to our MetaMask wallet

>    3. disable the button

Reviewer: ```suggestion
   3. Disable the button
```

Let's get to it!!

### MetaMask Extension Check

In our code we need to connect to our button from our index.html
``` javascript
const initialize = () => { 
  //Basic Actions Section
  const onboardButton = document.getElementById('connectButton')

}
```

Next we create a check function called `isMetaMaskInstalled` to see if the MetaMask extension is installed

``` javascript
const initialize = () => { 
  //Basic Actions Section
  const onboardButton = document.getElementById('connectButton')

  //Created check function to see if the MetaMask extenstion is installed
  const isMetaMaskInstalled = () => {

>     //Have to check the ehtereum binding on the window object to see if it's installed 

Reviewer: ```suggestion
    //Have to check the ethereum binding on the window object to see if it's installed 
```

### MetaMask "Installed" Dapp Flow

Next we need to revist our `MetamaskClientCheck` function and do similar functionality of what we did in our "MetaMask Not Installed" block to now our "MetaMask Is Installed" block of code.
``` javascript
const MetamaskClientCheck = () => {
    //Now we check to see if Metmask is installed
    if (!isMetaMaskInstalled()) {
      //If it isn't installed we ask the user to click to install it
      onboardButton.innerText = 'Click here to install MetaMask!'
      //When the button is clicked we call th is function
      onboardButton.onclick = onClickInstall
      //The button is now disabled
      onboardButton.disabled = false
    }
    else {
      //If MetaMask is installed we ask the user to connect to their wallet
      onboardButton.innerText = 'Connect'
      //When the button is clicked we call this function to connect the users MetaMask Wallet
      onboardButton.onclick = onClickConnect
      //The button is now disabled
      onboardButton.disabled = false
    }
  }
  MetamaskClientCheck();
```

> Now that we've created a function that will be called whenever we click the button to trigger a connection to our wallet and disabled the button. Let's create/dive into the `onClickConnect` function and build the logic inside of it.

Reviewer: ```suggestion
Now we've created a function that will be called whenever we click the button to trigger a connection to our wallet and disable the button. Next, let's dive into the `onClickConnect` function and build the logic inside of it.
```
